dropwizard-index-page/src/main/java/com/palantir/indexpage/IndexPageServlet.java

```diff
@@ -37,35 +33,43 @@
     private static final String CACHE_CONTROL = "no-cache, no-store, max-age=0, must-revalidate";
 
     @SuppressFBWarnings("SE_BAD_FIELD")
-    private final DefaultMustacheFactory factory;
-    private final File indexPage;
+    private final Asset indexPage;
 
+    /**
+     * Creates a new servlet that will try to serve the index page from the local file path. If the page doesn't exist
+     * on the disk, the servlet will try to load it from the classpath.
+     *
+     * @param indexPagePath
+     *        the path to the index path and it can either be the classpath or the local file path
+     * @throws IllegalArgumentException if the index page is not found
+     */
     public IndexPageServlet(String indexPagePath) {
         checkArgument(!Strings.isNullOrEmpty(indexPagePath));
 
-        this.factory = new DefaultMustacheFactory();
-        this.indexPage = new File(indexPagePath);
+        File file = new File(indexPagePath);
+        if (file.exists()) {
+            indexPage = new FileSystemAsset(file);
+        } else {
+            indexPage = new ClasspathAsset(Resources.getResource(indexPagePath));
+        }
     }
 
     @Override
     protected void doGet(HttpServletRequest request, HttpServletResponse response)
             throws ServletException, IOException {
 
-        response.setHeader(HttpHeaders.CACHE_CONTROL, CACHE_CONTROL);
-        response.setHeader(HttpHeaders.CONTENT_TYPE, MediaType.HTML_UTF_8.toString());
-
-        Map<String, String> context = ImmutableMap.of(BASE_URL, request.getContextPath() + "/");
-        BufferedReader reader;
-        try {
-            reader = Files.newReader(indexPage, Charsets.UTF_8);
-        } catch (FileNotFoundException e) {
+        Mustache template = indexPage.getTemplate();
+        if (template == null) {
             response.sendError(HttpStatus.NOT_FOUND_404, "Index page file not found.");
             return;
         }
 
-        Mustache mustache = factory.compile(reader, IndexPageBundle.INDEX_PAGE_NAME);
+        response.setHeader(HttpHeaders.CACHE_CONTROL, CACHE_CONTROL);
+        response.setHeader(HttpHeaders.CONTENT_TYPE, MediaType.HTML_UTF_8.toString());
+
+        Map<String, String> context = ImmutableMap.of(BASE_URL, request.getContextPath() + "/");
         PrintWriter writer = response.getWriter();
-        mustache.execute(writer, context);
+        template.execute(writer, context);
```

Comment:
this is still executing the templating logic on every request, right? isn't that what we're trying to avoid?